### PR TITLE
Add binary ops benchmark

### PR DESCRIPTION
**Pull request by karthikeyann**
closes https://github.com/rapidsai/cudf/issues/4118

### CODE REVIEW

File: cpp/benchmarks/binaryop/jit_binaryop_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+using namespace cudf;
+
+// Types from tests.
+// BINARYOP_BENCHMARK_DEFINE(timestamp_D,  duration_s,   ADD,                  timestamp_us);
+// BINARYOP_BENCHMARK_DEFINE(timestamp_D,  timestamp_s,  SUB,                  duration_s);
+// BINARYOP_BENCHMARK_DEFINE(duration_s,   int32_t,      MUL,                  duration_ns);
+// BINARYOP_BENCHMARK_DEFINE(duration_D,   duration_s,   DIV,                  int64_t);
+// BINARYOP_BENCHMARK_DEFINE(float,        float,        FLOOR_DIV,            int64_t);
+// BINARYOP_BENCHMARK_DEFINE(duration_D,   int64_t,      MOD,                  duration_us);
+// BINARYOP_BENCHMARK_DEFINE(duration_s,   duration_ms,  MOD,                  duration_us);
```
Comment (ttnghia) [inline cpp/benchmarks/binaryop/jit_binaryop_benchmark.cpp:77]: Should we clean this up?

File: cpp/benchmarks/binaryop/jit_binaryop_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/binaryop.hpp>
+
+#include <cudf_test/column_wrapper.hpp>
+
+#include <fixture/benchmark_fixture.hpp>
```
Comment (ttnghia) [inline cpp/benchmarks/binaryop/jit_binaryop_benchmark.cpp:21]: Please reorder the headers (close to far).

**Review by ttnghia — CHANGES_REQUESTED** [review]

**Review by ttnghia — APPROVED** [review]

**Review by rgsl888prabhu — APPROVED** [review]

**Comment by karthikeyann** [comment]
```
JIT_BINARYOP<float, int64_t, int32_t>/ADD/10000/manual_time                                                        0.017 ms        0.037 ms        41566
JIT_BINARYOP<float, int64_t, int32_t>/ADD/100000/manual_time                                                       0.022 ms        0.040 ms        31392
JIT_BINARYOP<float, int64_t, int32_t>/ADD/1000000/manual_time                                                      0.047 ms        0.064 ms        14925
JIT_BINARYOP<float, int64_t, int32_t>/ADD/10000000/manual_time                                                     0.280 ms        0.299 ms         2501
JIT_BINARYOP<float, int64_t, int32_t>/ADD/100000000/manual_time                                                     2.66 ms         2.69 ms          264
JIT_BINARYOP<duration_s, duration_D, duration_ms>/SUB/10000/manual_time                                            0.018 ms        0.037 ms        38226
JIT_BINARYOP<duration_s, duration_D, duration_ms>/SUB/100000/manual_time                                           0.022 ms        0.040 ms        29850
JIT_BINARYOP<duration_s, duration_D, duration_ms>/SUB/1000000/manual_time                                          0.051 ms        0.070 ms        13703
JIT_BINARYOP<duration_s, duration_D, duration_ms>/SUB/10000000/manual_time                                         0.345 ms        0.365 ms         2023
JIT_BINARYOP<duration_s, duration_D, duration_ms>/SUB/100000000/manual_time                                         3.41 ms         3.45 ms          205
JIT_BINARYOP<float, float, int64_t>/MUL/10000/manual_time                                                          0.019 ms        0.038 ms        34950
JIT_BINARYOP<float, float, int64_t>/MUL/100000/manual_time                                                         0.022 ms        0.039 ms        31819
JIT_BINARYOP<float, float, int64_t>/MUL/1000000/manual_time                                                        0.044 ms        0.064 ms        15740
JIT_BINARYOP<float, float, int64_t>/MUL/10000000/manual_time                                                       0.286 ms        0.306 ms         2444
JIT_BINARYOP<float, float, int64_t>/MUL/100000000/manual_time                                                       2.87 ms         2.90 ms          244
JIT_BINARYOP<int64_t, int64_t, int64_t>/DIV/10000/manual_time                                                      0.018 ms        0.037 ms        37241
JIT_BINARYOP<int64_t, int64_t, int64_t>/DIV/100000/manual_time                                                     0.024 ms        0.041 ms        28941
JIT_BINARYOP<int64_t, int64_t, int64_t>/DIV/1000000/manual_time                                                    0.057 ms        0.076 ms        12263
JIT_BINARYOP<int64_t, int64_t, int64_t>/DIV/10000000/manual_time                                                   0.408 ms        0.428 ms         1717
JIT_BINARYOP<int64_t, int64_t, int64_t>/DIV/100000000/manual_time                                                   3.96 ms         4.00 ms          177
JIT_BINARYOP<int64_t, int64_t, int64_t>/TRUE_DIV/10000/manual_time                                                 0.019 ms        0.038 ms        37909
JIT_BINARYOP<int64_t, int64_t, int64_t>/TRUE_DIV/100000/manual_time                                                0.023 ms        0.040 ms        30107
JIT_BINARYOP<int64_t, int64_t, int64_t>/TRUE_DIV/1000000/manual_time                                               0.056 ms        0.075 ms        12338
JIT_BINARYOP<int64_t, int64_t, int64_t>/TRUE_DIV/10000000/manual_time                                              0.406 ms        0.426 ms         1724
JIT_BINARYOP<int64_t, int64_t, int64_t>/TRUE_DIV/100000000/manual_time                                              3.96 ms         4.00 ms          177
JIT_BINARYOP<int64_t, int64_t, int64_t>/FLOOR_DIV/10000/manual_time                                                0.019 ms        0.038 ms        35681
JIT_BINARYOP<int64_t, int64_t, int64_t>/FLOOR_DIV/100000/manual_time                                               0.024 ms        0.041 ms        29291
JIT_BINARYOP<int64_t, int64_t, int64_t>/FLOOR_DIV/1000000/manual_time                                              0.057 ms        0.076 ms        12018
JIT_BINARYOP<int64_t, int64_t, int64_t>/FLOOR_DIV/10000000/manual_time                                             0.407 ms        0.427 ms         1717
JIT_BINARYOP<int64_t, int64_t, int64_t>/FLOOR_DIV/100000000/manual_time                                             3.96 ms         3.99 ms          177
JIT_BINARYOP<double, double, double>/MOD/10000/manual_time                                                         0.018 ms        0.038 ms        38669
JIT_BINARYOP<double, double, double>/MOD/100000/manual_time                                                        0.023 ms        0.041 ms        28886
JIT_BINARYOP<double, double, double>/MOD/1000000/manual_time                                                       0.058 ms        0.076 ms        11825
JIT_BINARYOP<double, double, double>/MOD/10000000/manual_time                                                      0.408 ms        0.427 ms         1717
JIT_BINARYOP<double, double, double>/MOD/100000000/manual_time                                                      3.97 ms         4.02 ms          176
JIT_BINARYOP<int64_t, int64_t, double>/POW/10000/manual_time                                                       0.019 ms        0.038 ms        35454
JIT_BINARYOP<int64_t, int64_t, double>/POW/100000/manual_time                                                      0.026 ms        0.044 ms        26382
JIT_BINARYOP<int64_t, int64_t, double>/POW/1000000/manual_time                                                     0.062 ms        0.081 ms        10983
JIT_BINARYOP<int64_t, int64_t, double>/POW/10000000/manual_time                                                    0.422 ms        0.441 ms         1662
JIT_BINARYOP<int64_t, int64_t, double>/POW/100000000/manual_time                                                    4.06 ms         4.10 ms          172
JIT_BINARYOP<int64_t, int32_t, int16_t>/BITWISE_AND/10000/manual_time                                              0.016 ms        0.036 ms        42023
JIT_BINARYOP<int64_t, int32_t, int16_t>/BITWISE_AND/100000/manual_time                                             0.021 ms        0.039 ms        32868
JIT_BINARYOP<int64_t, int32_t, int16_t>/BITWISE_AND/1000000/manual_time                                            0.046 ms        0.062 ms        15250
JIT_BINARYOP<int64_t, int32_t, int16_t>/BITWISE_AND/10000000/manual_time                                           0.258 ms        0.275 ms         2716
JIT_BINARYOP<int64_t, int32_t, int16_t>/BITWISE_AND/100000000/manual_time                                           2.40 ms         2.42 ms          292
JIT_BINARYOP<int16_t, int32_t, int64_t>/BITWISE_OR/10000/manual_time                                               0.017 ms        0.036 ms        40212
JIT_BINARYOP<int16_t, int32_t, int64_t>/BITWISE_OR/100000/manual_time                                              0.020 ms        0.038 ms        32908
JIT_BINARYOP<int16_t, int32_t, int64_t>/BITWISE_OR/1000000/manual_time                                             0.039 ms        0.059 ms        17544
JIT_BINARYOP<int16_t, int32_t, int64_t>/BITWISE_OR/10000000/manual_time                                            0.253 ms        0.273 ms         2711
JIT_BINARYOP<int16_t, int32_t, int64_t>/BITWISE_OR/100000000/manual_time                                            2.57 ms         2.59 ms          273
JIT_BINARYOP<int16_t, int64_t, int32_t>/BITWISE_XOR/10000/manual_time                                              0.018 ms        0.037 ms        38144
JIT_BINARYOP<int16_t, int64_t, int32_t>/BITWISE_XOR/100000/manual_time                                             0.021 ms        0.039 ms        32280
JIT_BINARYOP<int16_t, int64_t, int32_t>/BITWISE_XOR/1000000/manual_time                                            0.044 ms        0.060 ms        15996
JIT_BINARYOP<int16_t, int64_t, int32_t>/BITWISE_XOR/10000000/manual_time                                           0.247 ms        0.266 ms         2843
JIT_BINARYOP<int16_t, int64_t, int32_t>/BITWISE_XOR/100000000/manual_time                                           2.33 ms         2.36 ms          300
JIT_BINARYOP<double, int8_t, int16_t>/LOGICAL_AND/10000/manual_time                                                0.017 ms        0.037 ms        38241
JIT_BINARYOP<double, int8_t, int16_t>/LOGICAL_AND/100000/manual_time                                               0.019 ms        0.038 ms        35425
JIT_BINARYOP<double, int8_t, int16_t>/LOGICAL_AND/1000000/manual_time                                              0.040 ms        0.056 ms        17415
JIT_BINARYOP<double, int8_t, int16_t>/LOGICAL_AND/10000000/manual_time                                             0.206 ms        0.223 ms         3385
JIT_BINARYOP<double, int8_t, int16_t>/LOGICAL_AND/100000000/manual_time                                             1.90 ms         1.92 ms          369
JIT_BINARYOP<int16_t, int64_t, bool>/LOGICAL_OR/10000/manual_time                                                  0.018 ms        0.037 ms        38691
JIT_BINARYOP<int16_t, int64_t, bool>/LOGICAL_OR/100000/manual_time                                                 0.021 ms        0.039 ms        32616
JIT_BINARYOP<int16_t, int64_t, bool>/LOGICAL_OR/1000000/manual_time                                                0.044 ms        0.059 ms        15821
JIT_BINARYOP<int16_t, int64_t, bool>/LOGICAL_OR/10000000/manual_time                                               0.212 ms        0.228 ms         3309
JIT_BINARYOP<int16_t, int64_t, bool>/LOGICAL_OR/100000000/manual_time                                               1.95 ms         1.97 ms          359
JIT_BINARYOP<timestamp_s, timestamp_s, bool>/LESS/10000/manual_time                                                0.017 ms        0.036 ms        40129
JIT_BINARYOP<timestamp_s, timestamp_s, bool>/LESS/100000/manual_time                                               0.023 ms        0.041 ms        30412
JIT_BINARYOP<timestamp_s, timestamp_s, bool>/LESS/1000000/manual_time                                              0.055 ms        0.071 ms        12317
JIT_BINARYOP<timestamp_s, timestamp_s, bool>/LESS/10000000/manual_time                                             0.311 ms        0.327 ms         2252
JIT_BINARYOP<timestamp_s, timestamp_s, bool>/LESS/100000000/manual_time                                             2.92 ms         2.95 ms          240
JIT_BINARYOP<timestamp_ms, timestamp_s, bool>/GREATER/10000/manual_time                                            0.017 ms        0.036 ms        39902
JIT_BINARYOP<timestamp_ms, timestamp_s, bool>/GREATER/100000/manual_time                                           0.022 ms        0.041 ms        30279
JIT_BINARYOP<timestamp_ms, timestamp_s, bool>/GREATER/1000000/manual_time                                          0.055 ms        0.070 ms        12281
JIT_BINARYOP<timestamp_ms, timestamp_s, bool>/GREATER/10000000/manual_time                                         0.311 ms        0.328 ms         2229
JIT_BINARYOP<timestamp_ms, timestamp_s, bool>/GREATER/100000000/manual_time                                         2.96 ms         2.98 ms          237
JIT_BINARYOP<int, int, int>/SHIFT_LEFT/10000/manual_time                                                           0.018 ms        0.037 ms        37767
JIT_BINARYOP<int, int, int>/SHIFT_LEFT/100000/manual_time                                                          0.021 ms        0.039 ms        33026
JIT_BINARYOP<int, int, int>/SHIFT_LEFT/1000000/manual_time                                                         0.040 ms        0.056 ms        17527
JIT_BINARYOP<int, int, int>/SHIFT_LEFT/10000000/manual_time                                                        0.216 ms        0.235 ms         3244
JIT_BINARYOP<int, int, int>/SHIFT_LEFT/100000000/manual_time                                                        2.09 ms         2.11 ms          336
JIT_BINARYOP<int16_t, int64_t, int>/SHIFT_RIGHT/10000/manual_time                                                  0.017 ms        0.036 ms        41459
JIT_BINARYOP<int16_t, int64_t, int>/SHIFT_RIGHT/100000/manual_time                                                 0.021 ms        0.039 ms        33895
JIT_BINARYOP<int16_t, int64_t, int>/SHIFT_RIGHT/1000000/manual_time                                                0.044 ms        0.060 ms        15244
JIT_BINARYOP<int16_t, int64_t, int>/SHIFT_RIGHT/10000000/manual_time                                               0.246 ms        0.265 ms         2845
JIT_BINARYOP<int16_t, int64_t, int>/SHIFT_RIGHT/100000000/manual_time                                               2.33 ms         2.36 ms          300
JIT_BINARYOP<int64_t, int32_t, int64_t>/SHIFT_RIGHT_UNSIGNED/10000/manual_time                                     0.017 ms        0.036 ms        40319
JIT_BINARYOP<int64_t, int32_t, int64_t>/SHIFT_RIGHT_UNSIGNED/100000/manual_time                                    0.022 ms        0.040 ms        31175
JIT_BINARYOP<int64_t, int32_t, int64_t>/SHIFT_RIGHT_UNSIGNED/1000000/manual_time                                   0.050 ms        0.069 ms        13064
JIT_BINARYOP<int64_t, int32_t, int64_t>/SHIFT_RIGHT_UNSIGNED/10000000/manual_time                                  0.344 ms        0.364 ms         2024
JIT_BINARYOP<int64_t, int32_t, int64_t>/SHIFT_RIGHT_UNSIGNED/100000000/manual_time                                  3.38 ms         3.42 ms          207
JIT_BINARYOP<int32_t, int64_t, double>/PMOD/10000/manual_time                                                      0.018 ms        0.037 ms        38752
JIT_BINARYOP<int32_t, int64_t, double>/PMOD/100000/manual_time                                                     0.023 ms        0.041 ms        29659
JIT_BINARYOP<int32_t, int64_t, double>/PMOD/1000000/manual_time                                                    0.051 ms        0.070 ms        13056
JIT_BINARYOP<int32_t, int64_t, double>/PMOD/10000000/manual_time                                                   0.346 ms        0.366 ms         2021
JIT_BINARYOP<int32_t, int64_t, double>/PMOD/100000000/manual_time                                                   3.45 ms         3.49 ms          203
JIT_BINARYOP<float, double, double>/ATAN2/10000/manual_time                                                        0.018 ms        0.037 ms        37912
JIT_BINARYOP<float, double, double>/ATAN2/100000/manual_time                                                       0.023 ms        0.040 ms        29701
JIT_BINARYOP<float, double, double>/ATAN2/1000000/manual_time                                                      0.051 ms        0.070 ms        13490
JIT_BINARYOP<float, double, double>/ATAN2/10000000/manual_time                                                     0.348 ms        0.367 ms         2016
JIT_BINARYOP<float, double, double>/ATAN2/100000000/manual_time                                                     3.49 ms         3.53 ms          201
```

**Review by harrism — APPROVED** [review]

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**